sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/implementation/ManagementLocksImpl.java

```diff
@@ -0,0 +1,221 @@
+    }
+
+    @Override
+    public PagedFlux<ManagementLock> listByResourceGroupAsync(String resourceGroupName) {
+        return wrapPageAsync(this.manager().managementLockClient().getManagementLocks()
+            .listByResourceGroupAsync(resourceGroupName));
+    }
+
+    @Override
+    public ManagementLock getByResourceGroup(String resourceGroupName, String name) {
+        return this.getByResourceGroupAsync(resourceGroupName, name).block();
+    }
+
+    @Override
+    public Mono<ManagementLock> getByResourceGroupAsync(String resourceGroupName, String name) {
+        return this.manager().managementLockClient().getManagementLocks()
+            .getByResourceGroupAsync(resourceGroupName, name)
+            .map(this::wrapModel);
+    }
+
+    @Override
+    public ManagementLock getById(String id) {
+        return this.getByIdAsync(id).block();
+    }
+
+    @Override
+    public Mono<ManagementLock> getByIdAsync(String id) {
+        String resourceId = resourceIdFromLockId(id);
+        String lockName = ResourceUtils.nameFromResourceId(id);
+        return this.manager().managementLockClient().getManagementLocks().getByScopeAsync(resourceId, lockName)
+            .map(this::wrapModel);
+    }
+
+    @Override
+    public void deleteByResourceGroup(String resourceGroupName, String name) {
+        this.deleteByResourceGroupAsync(resourceGroupName, name).block();
+    }
+
+    @Override
+    public Mono<Void> deleteByResourceGroupAsync(String resourceGroupName, String name) {
+        return this.manager().managementLockClient().getManagementLocks()
+            .deleteAtResourceGroupLevelAsync(resourceGroupName, name);
+    }
+
+    @Override
+    public Flux<String> deleteByIdsAsync(Collection<String> ids) {
+        if (ids == null || ids.isEmpty()) {
+            return Flux.empty();
+        }
+
+        return Flux.fromIterable(ids)
+            .flatMapDelayError(id -> {
+                String lockName = ResourceUtils.nameFromResourceId(id);
+                String scopeName = ManagementLocksImpl.resourceIdFromLockId(id);
+                return this.manager().managementLockClient().getManagementLocks()
+                    .deleteByScopeAsync(scopeName, lockName)
+                    .then(Mono.just(id));
+            }, 32, 32);
+    }
+
+    @Override
+    public Flux<String> deleteByIdsAsync(String... ids) {
+        return this.deleteByIdsAsync(new ArrayList<>(Arrays.asList(ids)));
+    }
+
+    @Override
+    public void deleteByIds(Collection<String> ids) {
+        this.deleteByIdsAsync(ids).blockLast();
+    }
+
+    @Override
+    public void deleteByIds(String... ids) {
+        this.deleteByIdsAsync(ids).blockLast();
+    }
+
+    @Override
+    public ResourceManager manager() {
+        return this.manager;
+    }
+
+    @Override
+    public PagedIterable<ManagementLock> listForResource(String resourceId) {
+        return wrapList(this.manager().managementLockClient().getManagementLocks().listAtResourceLevel(
```

Comment:
Seems `listByScope` is easier?

Reply:
By testing, `listByScope` should be the right one as parsing id would fail if it is resource group id here.